peer.go

```diff
@@ -175,8 +186,8 @@ func (l *PeerList) Remove(hostPort string) error {
 	return nil
 }
 func (l *PeerList) choosePeer(prevSelected map[string]struct{}, avoidHost bool) *Peer {
-	var psPopList []*peerScore
-	var ps *peerScore
+	var chosenPSList []*peerScore
```

Comment:
presize chosenPSList to avoid a bunch of allocations on each selection, especially when it's > 1